src/registrar/management/commands/populate_domain_request_dates.py

```diff
@@ -0,0 +1,42 @@
+import logging
+from django.core.management import BaseCommand
+from registrar.management.commands.utility.terminal_helper import PopulateScriptTemplate, TerminalColors
+from registrar.models import DomainRequest
+from auditlog.models import LogEntry
+from django.core.exceptions import ObjectDoesNotExist
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand, PopulateScriptTemplate):
+    help = "Loops through each domain request object and populates the last_status_update and first_submitted_date"
+
+    def handle(self, **kwargs):
+        """Loops through each DomainRequest object and populates its last_status_update and first_submitted_date values"""
+        self.mass_update_records(DomainRequest, None, ["last_status_update", "first_submitted_date"])
+
+    def update_record(self, record: DomainRequest):
+        """Defines how we update the first_submitted_date and last_status_update fields"""
+        try:
+            # Retrieve and order audit log entries by timestamp in descending order
+            audit_log_entries = LogEntry.objects.filter(object_pk=record.pk).order_by("-timestamp")
+
+            # Loop through logs in descending order to find most recent status change
+            for log_entry in audit_log_entries:
+                if "status" in log_entry.changes:
```

Comment:
I think you're right, checking for 'status' in the changes_dict would be more robust as it will check for an actual element instead of just the string anywhere